use-case-django-continuous-integration.md

```diff
@@ -0,0 +1,148 @@
+blocks, one for setup and two for tests:
+
+<pre><code class="language-yaml"># .semaphore/semaphore.yml
+version: v1.0
+name: Example Django Application
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+blocks:
+  - name: Setup
+    task:
+      jobs:
+        - name: Cache pip downloads
+          commands:
+            - sem-version python 3.7
+            - checkout
+            - cache restore requirements-$(checksum requirements.txt)
+            - pip download --cache-dir .pip_cache -r requirements.txt
+            - cache store requirements-$(checksum requirements.txt) .pip_cache
+
+  - name: Lint
+    task:
+      prologue:
+        commands:
+          - sem-version python 3.7
+          - checkout
+          - cache restore requirements-$(checksum requirements.txt)
+          - pip install -r requirements.txt --cache-dir .pip_cache
+      jobs:
+        - name: Project
+          commands:
+            # NOTE: The oddss are you'll need to rework the file selection to fit your project
+            - git ls-files | grep -v 'migrations' | grep -v 'settings.py' | grep -v 'manage.py' | grep -E '.py$' | xargs pylint
+
+  - name: Test
+    task:
+      env_vars:
+        - name: DJANGO_SETTINGS_MODULE
+          value: blog.ci
+      prologue:
+        commands:
+          - sem-service start postgres
+          - sem-version python 3.7
+          - checkout
+          - cache restore requirements-$(checksum requirements.txt)
+          - pip install -r requirements.txt --cache-dir .pip_cache
+      jobs:
+        # Create a job for each application in the project
+        - name: Posts App
+          commands:
+            - coverage run manage.py test posts
+            - coverage report
+        - name: Comments Apps
+          commands:
+            - coverage run manage.py test comments
+            - coverage report
+</code></pre>
+
+This example assumes a separate settings module to configure the
+database. The settings are applied by setting the
+`DJANGO_SETTINGS_MODULE` environemnt variable. Here's an example for
+configuring PostreSQL:
+
+<pre><code class="language-yaml"># blog/ci.py
+from .settings import *
+
+DATABASES['default'] = {
+    'ENGINE': 'django.db.backends.postgresql',
+    'USER': 'postgres',
+    'NAME': 'postgres',
+    'HOST': 'localhost'
+}
+</code></pre>
+
+PostgreSQL instances run inside each job and can be accessed with a
+blank password. For more information on configuring database access,
+including tips for Python, see the [Python language guide][python-guide].
+
+Firefox, Chrome, and Chrome Headless drivers for Capybara work out of the box,
+so you will not need to make any adjustment for browser tests to work on
+Semaphore.
+
+This example configures coverage.py and and pylint. Be sure to exclude
+the relevant files from coverage by creating a `.coveragerc`:
+
+<pre><code class="language-conf"># .coveragerc
+[run]
+source = .
+omit =
+  venv/*
+  manage.py
+  */wsgi.py
```

Comment:
Do we need to escape these for proper parsing?

Reply:
Shouldn't be necessary. I'll check after publishing.